src/core/backbone_router/ndproxy_table.cpp

```diff
@@ -0,0 +1,232 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements Thread NdProxy Table management.
+ */
+
+#include "ndproxy_table.hpp"
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+
+#include <algorithm>
+
+#include "common/locator-getters.hpp"
+#include "common/logging.hpp"
+
+namespace ot {
+
+namespace BackboneRouter {
+
+void NdProxyTable::NdProxy::Init(const Ip6::InterfaceIdentifier &aAddressIid,
+                                 const Ip6::InterfaceIdentifier &aMeshLocalIid,
+                                 uint16_t                        aRloc16,
+                                 uint32_t                        aTimeSinceLastTransaction)
+{
+    OT_ASSERT(!mValid);
+
+    mValid        = true;
+    mAddressIid   = aAddressIid;
+    mMeshLocalIid = aMeshLocalIid;
+    Update(aRloc16, aTimeSinceLastTransaction);
+}
+
+void NdProxyTable::NdProxy::Update(uint16_t aRloc16, uint32_t aTimeSinceLastTransaction)
+{
+    OT_ASSERT(mValid);
+
+    mRloc16 = aRloc16;
+    aTimeSinceLastTransaction =
+        std::min(aTimeSinceLastTransaction, static_cast<uint32_t>(Mle::kTimeSinceLastTransactionMax));
+    mLastRegistrationTime = TimerMilli::GetNow() - TimeMilli::SecToMsec(aTimeSinceLastTransaction);
```

Comment:
`aTimeSinceLastTransaction` is the number seconds since the device registers its DUA according to the DUA requester. 
I think PBBR convert it to local timestamp so that it can always calculate the time since device registering its DUA later. 
It has nothing to do with which device (PBBR or DUA requester) has a larger local timestamp. 

Thoughts? @librasungirl